gateway/src/resty/http/proxy.lua

```diff
@@ -133,15 +31,94 @@ end
 local function connect(request, skip_https_connect)
     local httpc = http.new()
     local proxy_uri = find_proxy_url(request)
+    local uri = request.uri
+    local scheme = uri.scheme
+    local host = uri.host
+    local port = default_port(uri)
 
-    request.ssl_verify = request.options and request.options.ssl and request.options.ssl.verify
-    request.proxy = proxy_uri
+    -- set ssl_verify: lua-resty-http set ssl_verify to true by default if scheme is https, whereas
+    -- openresty treat nil as false, so we need to explicitly set ssl_verify to false if nil
+    local ssl_verify = request.options and request.options.ssl and request.options.ssl.verify or false
+
+    local options = {
+      scheme = scheme,
+      host = host,
+      port = port
+    }
+    if scheme == 'https' then
+        options.ssl_server_name = host
+        options.ssl_verify = ssl_verify
+    end
 
+    -- Connect via proxy
     if proxy_uri then
-        return connect_proxy(httpc, request, skip_https_connect)
+        if proxy_uri.scheme ~= 'http' then
+            return nil, 'proxy connection supports only http'
+        else
+            proxy_uri.port = default_port(proxy_uri)
+        end
+
+        -- Resolve the proxy IP/Port
+        local proxy_host, proxy_port = httpc:resolve(proxy_uri.host, proxy_uri.port)
```

Comment:
Why resolve, the spec says it should not be resolved.

Reply:
I see, that's the proxy hostname. APIcast resolves itself instead of relying to lua resty lib doing that.